### PR TITLE
Support all `^TaskLike` in `do!` and `let!` similar to F#'s `task`, this includes non-generic `Task` and `ValueTask` with `do!`

### LINKED ISSUE #43: Allow use of `Task`, as opposed to `Task<'T>` in `do!` statements
Currently, this is not possible:

```f#
taskSeq {
   do! Task.Delay 220  // fails, because this returns a Task, not a Task<unit>
}
```

Add necessary overloads for allowing this, similarly to how the `task` CE works.

Of note is that allowing `Task` can lead to unexpected behavior, as `Task<'T>` inherits from `Task`. However, we should strive to have parity with `task` in F#. to avoid surprises for users. Just like with normal `task`, we would still disallow `do! task { return 10 }`. This should be OK, as the F# compiler does not (yet) allow that kind of conversion.

### LINKED ISSUE #110: Implement `^TaskLike` similar to F#'s `task`
To support the same types F# Core supports with `task`, specifically in `let!` and `do!`, we'll need to implement the `^TaskLike` SRTP approach. This prevents ambiguities caused by `Task<'T> :> Task`. It also limits the needed overloads for `ValueTask<'T>` and `ValueTask`.

Related to #43, which can only be implemented by using this approach.

This change will unify the types supported between `task` and `taskSeq`, except for `async`, which should be added next (see #79).

### PR DESCRIPTION
**Pull request by abelbraaksma**
Fixes #43.
Fixes #110.

This basically allows:

```f#
let t = task { return 10 } :> Task  // ignore by casting to Task
let ts = taskSeq { do! t } // currently not supported, must be Task<unit>
let ts = taskSeq { do! Task.Delay 10 } // currently not supported
```

On top of that, this PR will add support for all of `^TaskLike` that F#'s `task` also supports. This change is necessary, as the above distinction isn't possible by adding an overload for the non-generic `Task` and `ValueTask`, for the simple reason that `Task<'T> :> Task`.

The solution chosen in F# was to have a `Bind` implementation that takes an SRTP parameter of `^TaskLike`, which has a `GetAwaiter() -> 'T when 'T :> ICriticalNotifyCompletion`, as follows:

```f#
member inline _.Bind< ^TaskLike, 'TResult1, 'TResult2, ^Awaiter, 'TOverall
    when ^TaskLike: (member GetAwaiter: unit -> ^Awaiter)
    and ^Awaiter :> ICriticalNotifyCompletion
    and ^Awaiter: (member get_IsCompleted: unit -> bool)
    and ^Awaiter: (member GetResult: unit -> 'TResult1)

    ... resumable code implementation ...
```

Because `Task<'T>` does have a `GetAwaiter` _on the type_, but also inherits `Task` which in turn implements the non-generic `TaskAwaiter`, a conflict arises causing an "ambiguous overload" error with just the code above. To solve this, a high-priority overload `Bind` for `Task<'T>` is added, which specifically pick the `Task<'T>.GetAwaiter: unit -> TaskAwaiter<'T>`, as opposed to `(Task<'T> :> Task).GetAwaiter: unit -> unit`.

Note that the non-generic `Task` only has the non-generic `GetResult: unit -> unit`, which fits the above signature where `'TResult1` becomes `unit`, and no ambiguity arises.

In other words, `TaskLike` are: `ValueTask`, `ValueTask<'T>`, `Task`. Not `TaskLike` per the above definition is `Task<'T>`, which has its own overload.

### CODE REVIEW

**Comment by abelbraaksma** [comment]
This PR will be part of v0.3.0.
